### PR TITLE
support paddle_yolov5s transform to MLIR

### PR DESCRIPTION
**Pull request by hlt-bkx**
adding some files(for example:paddleCoverter.py) to convert the model from paddle_yolov5s to MLIR

### CODE REVIEW

File: python/tools/model_runner.py [hunk 1]
```diff
@@ -264,6 +264,31 @@ def generate_onnx_with_all(onnx_file: str):
         os.remove(onnx_file)
         return dict(filter(lambda x: isinstance(x[1], np.ndarray), zip(output_keys, outs)))
 
+def paddle_inference(inputs : dict, paddle_file : str, dump_all : bool = True) -> dict:
+    import paddle
+    paddle.enable_static()
+    exe = paddle.static.Executor(paddle.CPUPlace())
+    [inference_program, feed_target_names, fetch_targets] = (
+        paddle.static.load_inference_model('model', exe))
```
Comment (zhengqianisme) [inline python/tools/model_runner.py:272]: 读的pdmodel路径是当前路径下的model.pdmodel,不是paddle_file输入的任意路径。

File: python/tools/model_runner.py [hunk 2]
```diff
@@ -264,6 +264,31 @@ def generate_onnx_with_all(onnx_file: str):
...
+    out_data = exe.run(
+        inference_program,
+        feed = feed_dict,
+        fetch_list = out_name
+    )
+    outputs = dict()
+    for i in range(len(out_name)):
+        outputs[out_name[i]] = out_data[i]
+
+    #返回的是一个字典：
```
[the remark this answers is not in the document]
Comment (hlt-bkx) [reply python/tools/model_runner.py:288]: 好的，我调整一下

File: python/tools/model_runner.py (hunk 1 above)
Comment (hlt-bkx) [reply python/tools/model_runner.py:272]: 好的 这个我也调整一下

File: python/transform/paddleConverter.py [hunk 3]
```diff
@@ -0,0 +1,1186 @@
...
+        self.model_name = model_name
+        self.weight_file = "{}_top_origin_weight.npz".format(model_name)
+        self.model = None
+        self.mlir  = None
+        self.node_name_mapping = {}
+        self.load_paddle_model(paddle_file,input_shapes,output_names)
+        self.init_MLIRImporter()
+        self.preprocess_args = {}
+        if 'channel_format' in preprocess_args:
+            if preprocess_args['channel_format'] != "none":
```
Comment (zhengqianisme) [inline python/transform/paddleConverter.py:72]: Traceback (most recent call last):
  File "/workspace/tpu-mlir/python/tools/model_transform.py", line 296, in <module>
    tool.model_transform(args.mlir, args.add_postprocess)
  File "/workspace/tpu-mlir/python/tools/model_transform.py", line 50, in model_transform
    self.converter.generate_mlir(mlir_origin)
  File "/workspace/tpu-mlir/python/transform/paddleConverter.py", line 395, in generate_mlir
    input_ = self.mlir.create_input_op(self.get_loc(_name),idx,self.preprocess_args)
  File "/workspace/tpu-mlir/python/transform/MLIRImporter.py", line 183, in create_input_op
    input_op = top.InputOp(**init_args)
TypeError: InputOp.__init__() got an unexpected keyword argument 'preprocess_list'

Comment (hlt-bkx) [reply python/transform/paddleConverter.py:72]: image
您好 请问是遇到这个问题吗，这个问题好像是因为导入paddle库的问题。

https://stackoverflow.com/questions/72133316/libssl-so-1-1-cannot-open-shared-object-file-no-such-file-or-directory
可以参考这个问题，或者在docker下面运行如下命令，更新libssl.so为1.1的版本。
$ mkdir $HOME/opt && cd $HOME/opt
# Download a supported openssl version. e.g., openssl-1.1.1o.tar.gz or openssl-1.1.1t.tar.gz
$ wget https://www.openssl.org/source/openssl-1.1.1o.tar.gz
$ tar -zxvf openssl-1.1.1o.tar.gz
$ cd openssl-1.1.1o
$ ./config && make && make test
$ mkdir $HOME/opt/lib
$ mv $HOME/opt/openssl-1.1.1o/libcrypto.so.1.1 $HOME/opt/lib/
$ mv $HOME/opt/openssl-1.1.1o/libssl.so.1.1 $HOME/opt/lib/
export LD_LIBRARY_PATH=$HOME/opt/lib:$LD_LIBRARY_PATH

Comment (zhengqianisme) [reply python/transform/paddleConverter.py:72]: 您好，不是这个问题。
运行： model_transform.py --model_name model --model_def ../regression/model/model.pdmodel --input_shapes [[1,3,640,640]] --mean 0.0,0.0,0.0 --scale 0.0039216,0.0039216,0.0039216 --keep_aspect_ratio --pixel_format rgb --output_names 311,457,587 --test_input ../regression/image/dog.jpg --test_result yolov5s_top_outputs.npz --mlir yolov5s.mlir
报错：
Traceback (most recent call last):
  File "/workspace/tpu-mlir/python/tools/model_transform.py", line 295, in <module>
    tool.model_transform(args.mlir, args.add_postprocess)
  File "/workspace/tpu-mlir/python/tools/model_transform.py", line 50, in model_transform
    self.converter.generate_mlir(mlir_origin)
  File "/workspace/tpu-mlir/python/transform/paddleConverter.py", line 395, in generate_mlir
    input_ = self.mlir.create_input_op(self.get_loc(_name),idx,self.preprocess_args)
  File "/workspace/tpu-mlir/python/transform/MLIRImporter.py", line 183, in create_input_op
    input_op = top.InputOp(**init_args)
TypeError: InputOp.__init__() got an unexpected keyword argument 'preprocess_list'

Comment (hlt-bkx) [reply python/transform/paddleConverter.py:72]: 不好意思，我这边重新更换一下环境运行一下，看看如何解决这个问题。

Comment (hlt-bkx) [reply python/transform/paddleConverter.py:85]: 您好，我在最新环境下运行后也报了您那个错误。然后我重新调整了里面的转换代码，已经重新提交了代码，目前我这边运行结果是可以了，麻烦您有空试一下

**Comment by zhengqianisme** [comment]
您好，在paddle转mlir后，没有准确选取需要compare的算子进行compare。检查两者的name是否能匹配。
[Running]: npz_tool.py compare yolov5s_top_outputs.npz paddle_yolov5s_ref_outputs.npz --tolerance 0.99,0.99 --except - -vv
0it [00:00, ?it/s]0 compared
0 passed
  0 equal, 0 close, 0 similar
0 failed
  0 not equal, 0 not similar
min_similiarity = (1.0, 1.0, inf)

**Comment by hlt-bkx** [comment]
好的，我补充一下

**Comment by zhengqianisme** [comment]
BM1684X INT8量化报错。
执行的是：model_deploy.py   --mlir yolov5s_pd.mlir   --quantize INT8   --chip bm1684x   --test_input paddle_yolov5s_in_f32.npz   --test_reference yolov5s_top_outputs.npz  --calibration_table ../regression/cali_tables/yolov5s_cali_table  --model yolov5s_1684x_int8.bmodel 

报错信息：tpuc-opt: /usr/local/include/llvm/Support/Casting.h:566: decltype(auto) llvm::cast(const From &) [To = mlir::quant::CalibratedQuantizedType, From = mlir::Type]: Assertion `isa<To>(Val) && "cast<Ty>() argument of incompatible type!"' failed.

File: python/tools/model_runner.py [hunk 4]
```diff
@@ -55,8 +54,8 @@ def model_inference(inputs: dict, model_file: str, dump_all = True) -> dict:
         chip = get_chip_from_model(model_file)
         # trick for runtime link chip cmodel
         lib_so = 'libcmodel_1684x.so'
-        if chip == 'BM1688' or chip == 'CV186X':
-            lib_so = 'libcmodel_1688.so'
+        if chip == 'BM1686' or chip == 'CV186X':
+            lib_so = 'libcmodel_1686.so'
```
Comment (zhengqianisme) [inline python/tools/model_runner.py:58]: if chip == 'BM1688' or chip == 'CV186X':
            lib_so = 'libcmodel_1688.so'

Comment (hlt-bkx) [reply python/tools/model_runner.py:58]: 你好，请问那个昨天的INT8量化问题后面解决了吗？然后请问是要在runner.py里面加入1688芯片的判别吗

**Pull request closed without merging**